generator/__init__.py

```diff
@@ -6,17 +6,36 @@
 
 __docformat__ = "restructuredtext"
 
+import sys
 import warnings
 
 import click
+from google.auth.exceptions import DefaultCredentialsError
+from google.cloud import bigquery
 
 from .lookml import lookml
 from .namespaces import namespaces
 from .spoke import update_spoke
 
 
+def is_authenticated():
+    """Check if the user is authenticated to GCP."""
+    try:
+        bigquery.Client()
+    except DefaultCredentialsError:
+        return False
+    return True
```

Comment:
I see this is using the [same `is_authenticated()` logic as BigQuery ETL](https://github.com/mozilla/bigquery-etl/blob/f86a8d7ac197ac32d65badc7228966877199c829/bigquery_etl/cli/utils.py#L42-L48), which is fine for now, but it has the same drawback where after Google application default credentials have been set at least once this will always return true, and at runtime you'll get a `google.auth.exceptions.RefreshError: Reauthentication is needed` error.

So if/when we eventually address that drawback for BigQuery ETL we should remember to do so here as well.